app/models/movie.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Movie < ApplicationRecord
+  has_many :genres
+  has_many :reviews, dependent: :destroy # If admin removes a movie from the database, reviews related to that movie, will also be removed
+
+  validates :title, :director, :synopsis, presence: true
+  validates :release_date, presence: true
+
+  def all_genres
+    Genre.for_movie(self)
+  end
+
+  def all_reviews_with_comments
+    reviews.includes(:comments)
```

Comment:
How does `.includes` work?

Reply:
Apu , i wanted to load all the associated comments while loading a review. 

At first i thought of displaying comments and reviews at the same time for a movie, so i wrote this method. 

When i write movie.all_reviews_with_comments , the. db loads all the reviews associated with the movie, and all the comments associated with the corresponding review  

As i moved forward with the project, i didnt use the method . I will remove it